### PR TITLE
BUG: Fix casting issue in python wrappings generated in SkeletalRepresentation

### LINKED ISSUE #142: Error while building SlicerSALT in docker
hi, 

I am trying to build SlicerSALT using the qt5-centos7 Slicer build environment Docker file but the compilation ends with errors. The errors occur while building SRep.

@jcfr, @phcerdan, do you know what could be the problem?

Here is the error I got:

```
[ 74%] Building CXX object E/SRep/SkeletalRepresentationInitializer/Logic/CMakeFiles/vtkSlicerSkeletalRepresentationInitializerModuleLogicPythonD.dir/vtkBackwardFlowLogicPython.cxx.o

/osm_qSlicerUnitsModuleWidgets_module_init.cpp.o
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx: In function ‘PyObject* PyvtkBackwardFlowLogic_runApplyTPS(PyObject*, PyObject*)’:
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx:25:68: error: invalid static_cast from type ‘vtkObjectBase*’ to type ‘vtkBackwardFlowLogic*’
   vtkBackwardFlowLogic *op = static_cast<vtkBackwardFlowLogic *>(vp);
                                                                    ^
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx: In function ‘PyObject* PyvtkBackwardFlowLogic_computePairwiseTPS(PyObject*, PyObject*)’:
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx:55:68: error: invalid static_cast from type ‘vtkObjectBase*’ to type ‘vtkBackwardFlowLogic*’
   vtkBackwardFlowLogic *op = static_cast<vtkBackwardFlowLogic *>(vp);
                                                                    ^
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx: In function ‘PyObject* PyvtkBackwardFlowLogic_generateEllipsoidSrep(PyObject*, PyObject*)’:
/NIRAL/work/lpzmateo/source/Projects/SlicerSALT-build/Slicer-build/E/SRep/SkeletalRepresentationInitializer/Logic/vtkBackwardFlowLogicPython.cxx:91:68: error: invalid static_cast from type ‘vtkObjectBase*’ to type ‘vtkBackwardFlowLogic*’
   vtkBackwardFlowLogic *op = static_cast<vtkBackwardFlowLogic *>(vp);
                                                                    ^
make[5]: *** [E/SRep/SkeletalRepresentationInitializer/Logic/CMakeFiles/vtkSlicerSkeletalRepresentationInitializerModuleLogicPythonD.dir/vtkBackwardFlowLogicPython.cxx.o] Error 1
make[4]: *** [E/SRep/SkeletalRepresentationInitializer/Logic/CMakeFiles/vtkSlicerSkeletalRepresentationInitializerModuleLogicPythonD.dir/all] Error 2
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [all] Error 2
make[2]: *** [slicersources-build/Slicer-prefix/src/Slicer-stamp/Slicer-build] Error 2
make[1]: *** [slicersources-build/CMakeFiles/Slicer.dir/all] Error 2
make: *** [all] Error 2
```

### PR DESCRIPTION
**Pull request by phcerdan**
BUG: Fix casting issue in python wrappings generated in SkeletalRepresentation

Fix #142

Update SkeletalRepresentation to master.

```
git shortlog --no-merges 027a1b8a..7be23feb

Pablo Hernandez-Cerdan (3):
      COMP: Fix warnings in vtkBackwardFlowLogic
      STYLE: Remove unused macros, fix indent
      COMP: Avoid running the mock test in SkeletalRepresentationVisualizer

lopezmt (1):
      FIX: casting issue with vtkBackwardFlowLogic and vtkForwardFlowLogic
```

Also, use SRep as a short_extension_name to avoid long path problems on Windows.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -196,11 +196,11 @@ list(APPEND Slicer_EXTENSION_SOURCE_DIRS ${${extension_name}_SOURCE_DIR})
 #-----------------------------------------------------------------------------
 # Srep
 set(extension_name "SlicerSkeletalRepresentation")
-set(${extension_name}_SOURCE_DIR "${CMAKE_BINARY_DIR}/SRep")
+set(${extension_name}_SOURCE_DIR "${CMAKE_BINARY_DIR}/${extension_name}")
```
Comment (jcfr) [inline CMakeLists.txt]: I suggest to keep the extension name short. Path length could then easily become an issue.

**Comment by phcerdan** [comment]
Thanks @jcfr , change the extension name to SRep then.

File: CMakeLists.txt [hunk 2]
```diff
@@ -195,12 +195,12 @@ list(APPEND Slicer_EXTENSION_SOURCE_DIRS ${${extension_name}_SOURCE_DIR})
 
 #-----------------------------------------------------------------------------
 # Srep
-set(extension_name "SlicerSkeletalRepresentation")
-set(${extension_name}_SOURCE_DIR "${CMAKE_BINARY_DIR}/SRep")
+set(extension_name "SRep")
```
Comment (jcfr) [inline CMakeLists.txt]: Almost, the name of the extension is still `SlicerSkeletalRepresentation` but the source folder is `SRep`. I suggest to simply keep the file how it was.

Comment (jcfr) [reply CMakeLists.txt]: You could add a comment like 

```
set(${extension_name}_SOURCE_DIR "${CMAKE_BINARY_DIR}/SRep") # Use shorter extension name to issues with long paths on windows
```

Comment (phcerdan) [reply CMakeLists.txt]: Before this patch we have long names plus inconsistency.
image
image

 I suggest we take one of the both approaches on this patch, with your comment.

Comment (phcerdan) [reply CMakeLists.txt]: I added a short_extension_name, and use it for the FetchContent function and the `_SOURCE_DIR`. Let me know what you think. Also added the comment explaining why

Comment (jcfr) [reply CMakeLists.txt]: Well done :+1:

**Pull request merged**